superset/assets/visualizations/table.js

```diff
@@ -90,7 +90,8 @@ function tableVis(slice, payload) {
         html = `<span class="like-pre">${val}</span>`;
       }
       if (isMetric) {
-        html = slice.d3format(c, val);
+        const format = slice.datasource.column_formats && slice.datasource.column_formats[c] || fd.number_format || '.3s';
+        html = d3.format(format)(val);
```

Comment:
`utils.d3format` instead of `d3.format` does some caching/reuse of the formatter functions the it has seen in the past and should be more efficient.